### PR TITLE
WT-6351 Fix race between threads to update condition variable's previous wait value.

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
Here we are trying to read the `prev_wait` of condition variables's value using atomic operations because there can be multiple threads accessing this block of code simultaneously. 

The `cond->prev_wait` value was going above WT_MILLION due to race between threads and causing it to assert.

### CODE REVIEW

File: src/support/cond_auto.c [hunk 1]
```diff
@@ -55,7 +55,16 @@ __wt_cond_auto_wait_signal(WT_SESSION_IMPL *session, WT_CONDVAR *cond, bool prog
...
     else {
         delta = WT_MAX(1, (cond->max_wait - cond->min_wait) / 10);
+        /*
+         * Try to update the previous wait value for the condition variable. There can be multiple
+         * threads doing this concurrently, so use atomic operations to make sure the value remains
+         * within the bounds of the maximum configured. Don't retry if our update didn't make it in
+         * - it's not necessary for the previous wait time to be updated every time.
+         */
+        WT_ORDERED_READ(saved_prev_wait, cond->prev_wait);
         cond->prev_wait = WT_MIN(cond->max_wait, cond->prev_wait + delta);
```
Comment (sulabhM) [inline src/support/cond_auto.c:65]: We can still race on line 65 and end up with cond->prev_wait more than 1 million, like before the change. In such a case atomic compare on the next line will not fix this value and hence we will get the same bug as before. 

If you remove line 65, I think you fix the bug. order read assures the latest value in the local variable. Atomic compare makes sure that in the case of a race, only the first thread gets to modify successfully, the second thread leaves the value untouched. WT_MIN calculation from the local variable calculates the correct final value.

So i think the change should be :

```
+ WT_ORDERED_READ(saved_prev_wait, cond->prev_wait);
+        __wt_atomic_cas64(
+          &cond->prev_wait, saved_prev_wait, WT_MIN(cond->max_wait, saved_prev_wait + delta));
```

Comment (agorrod) [reply src/support/cond_auto.c:65]: +1

Comment (raviprakashgiri29) [reply src/support/cond_auto.c:65]: @agorrod : Yes, I agree with this and discussed it with @sulabhM.  I will do the changes soon.

**Comment by tammybailey** [comment]
@raviprakashgiri29 would you run a stress test against your latest changes, please? ty!

**Comment by raviprakashgiri29** [comment]
> @raviprakashgiri29 would you run a stress test against your latest changes, please? ty!

@tammybailey , Yes, I ran the stress test with the previous changes, but anyways I have created another patch build with the latest changes. Updated the link in the Jira ticket.

**Pull request merged**